### PR TITLE
feat(error): update error interface

### PR DESCRIPTION
**Pull request by szlee8**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -30,6 +30,23 @@ export interface IWorkingRetrieval {
...
+  error: IError
+}
+
+/**
+ *  Genric error interface that can be extended
+ */
+export interface IError {
+  statusCode: number;
+  serviceError?: IServiceError;
+  correlationVector ?: string;
```
Comment (connor4312) [inline src/index.ts]: Can you run npm run fmt?

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@mixer/retrieval",
-  "version": "1.0.0",
+  "version": "1.0.1",
```
Comment (connor4312) [inline package.json]: This is a breaking change, the version should be revved to 2.0.0

File: src/index.ts (hunk 1 above)
Comment (szlee8) [reply src/index.ts]: done

File: package.json (hunk 2 above)
Comment (szlee8) [reply package.json]: done

**Review by connor4312 — APPROVED** [review]

**Pull request merged**